src/Services/AllergyIntoleranceService.php

```diff
@@ -95,64 +187,21 @@ public function getAll($search = array(), $isAndCondition = true, $puuidBind = n
-        users.uuid as practitioner,
-        facility.uuid as organization,
-        patient.uuid as puuid,
-        reaction.title as reaction_title,
-        verification.title as verification_title
-    FROM lists
-        LEFT JOIN list_options as reaction ON (reaction.option_id = lists.reaction and reaction.list_id = 'reaction')
-        LEFT JOIN list_options as verification ON verification.option_id = lists.verification and verification.list_id = 'allergyintolerance-verification'
-        RIGHT JOIN patient_data as patient ON patient.pid = lists.pid
-        LEFT JOIN users as users ON users.username = lists.user
-        LEFT JOIN facility as facility ON facility.name = users.facility
-    WHERE type = 'allergy'";
-
-        if (!empty($search)) {
-            $sql .= ' AND ';
-            if (!empty($puuidBind)) {
-                // code to support patient binding
-                $sql .= '(';
-            }
-            $whereClauses = array();
-            foreach ($search as $fieldName => $fieldValue) {
-                array_push($whereClauses, $fieldName . ' = ?');
-                array_push($sqlBindArray, $fieldValue);
+        $newSearch = [];
+        foreach ($search as $key => $value) {
+            if (!$value instanceof ISearchField) {
+                $newSearch[] = new StringSearchField($key, [$value], SearchModifier::EXACT);
+            } else {
+                $newSearch[$key] = $value;
             }
-            $sqlCondition = ($isAndCondition == true) ? 'AND' : 'OR';
-            $sql .= implode(' ' . $sqlCondition . ' ', $whereClauses);
-            if (!empty($puuidBind)) {
-                // code to support patient binding
-                $sql .= ") AND `patient`.`uuid` = ?";
-                $sqlBindArray[] = UuidRegistry::uuidToBytes($puuidBind);
-            }
-        } elseif (!empty($puuidBind)) {
-            // code to support patient binding
-            $sql .= " AND `patient`.`uuid` = ?";
-            $sqlBindArray[] = UuidRegistry::uuidToBytes($puuidBind);
         }
 
-        $statementResults = sqlStatement($sql, $sqlBindArray);
-
-        $processingResult = new ProcessingResult();
-        while ($row = sqlFetchArray($statementResults)) {
-            $row['uuid'] = UuidRegistry::uuidToString($row['uuid']);
-            $row['puuid'] = UuidRegistry::uuidToString($row['puuid']);
-            $row['practitioner'] = $row['practitioner'] ?
-                UuidRegistry::uuidToString($row['practitioner']) :
-                $row['practitioner'];
-            $row['organization'] = $row['organization'] ?
-                UuidRegistry::uuidToString($row['organization']) :
-            $row['organization'];
-            if ($row['diagnosis'] != "") {
-                $row['diagnosis'] = $this->addCoding($row['diagnosis']);
-            }
-            $processingResult->addData($row);
+        // override our type and our puuid
+        if (isset($puuidBind) && !isset($search['puuid'])) {
```

Comment:
this if is throwing me off a bit. doesn't this mean that the bootstrapped patient can be overrided to another patient's data if they put another patient's uuid in the lists.pid search field?

Reply:
Good catch.  I've gone in and fixed that.  When I wrote the code I figured the puuidBind was already being set higher up in the FHIRAllergyIntolerance controller and I didn't want to overwrite it.  That's not a good thing here in this case.  That line should be fixed in the latest push I just sent up.